core-services/sanitize-prow-jobs/_config.yaml

```diff
@@ -1,6 +1,10 @@
 default: api.ci
 nonKubernetes: api.ci
 groups:
+  "app.ci":
+    jobs:
+    #name: app.ci-jobs
```

Comment:
Making a new comment because I can not unresolve the other one:

> I want to keep it there.
> For each group, we leave a comment #name: ... to remind me of what the group is about.
> We had name as a field of the json but it did not go through review. :)

I don't understand. On line four, there is the name `app.ci`. You are afraid you forgot about that when you get to line six so you add a comment there?

Reply:
Jobs could end up with `app.ci` by different reasons.

```
  "app.ci":
    jobs:
    #name: app.ci-jobs (has to be on api.ci because e.g., no required resource like SA on other clusters
    - job1
    - job2
    #name: the first 20% of total job volume
    - job3
    - job4
```

I use the comment as separation of GROUPs of jobs. I wanted `groups` under `jobs` and each group has a name. The comment is useful only for human.

Reply:
Okay, now we are getting somewhere. If you feel such a description is needed, please add it as a comment to each individual job and make it more precise, because you are the only person in this team that will understand the current meaning of the comment. Probably something like `Has to be on app.ci because the bitwarden secret is only available there due to its sensitivity`.

I don't think a comment like `The first 20% ...` is needed and will make things confusing in the future (at some point whats the first 20% today might be the first 5%). IMHO there is no reason to have comments for jobs that could run on any cluster.

Reply:
Comment added.

> I don't think a comment like The first 20% ... is needed and will make things confusing in the future (at some point whats the first 20% today might be the first 5%). IMHO there is no reason to have comments for jobs that could run on any cluster.

I agree. We could regenerate them in the future if we feel the current grouping is outdated and no longer accurate.
But that is not the scope of current PR, right?

Ideally, we should just migrate all of them, by one shot.
But as you have seen, we crashed build01 so many times.
Migration by groups sounds reasonable when we want to feel how far we have gone through the migration.

Reply:
Please do not start the comment off with the name of the cluster. The name of the cluster is the group name already, it doesn't get more obvious than that.

If you think further explanation is needed, please add a comment on the individual _job_, not on a group of some jobs. A comment like `These jobs have to run on app.ci for some reason` is useless, its obvious already by the fact that they are in the `app.ci` group.


Regarding the groupwise migration its clear why we do it like that which is why I don't think a comment like `first 20% of jobs` adds any value.

Reply:
It feels to me like what you are putting into the description should probably just be the group name.

Reply:
I need a way of dividing the jobs inside one cluster.
We do not have a `name` field in the config and that is why i put it into comment.

Is it getting any better now?

Reply:
yeah thats why I suggested the name, there is nothing that prevents us from having multiple groups for the same cluster.

Reply:
Sadly `group` did not get though the review.
I thought we could still use `name: ...` in the yaml even though it is not in the go code of `config` struct. An unused field has a better position than a comment. But anyway ...